### ISSUE TITLE
when step docker run it error ...

### ISSUE THREAD

**Issue by ericvlog** [akarazniewicz/stremio-jellyfin#1]
I install the requirement on jellyfin without problem , but when i run docker run ghcr.io/akarazniewicz/stremio-jellyfin -p 60421:60421 -e JELLYFIN_USER=plsharevme -e JELLYFIN_PASSWORD=1234567 -e JELLYFIN_SERVER=http://192.168.1.100:8096

the log files : 

              at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1602:16) {
            errno: -111,
            code: 'ECONNREFUSED',
            syscall: 'connect',
            address: '127.0.0.1',
            port: 80
          }
        },
        allowHalfOpen: false,
        _maxListeners: undefined,
        _eventsCount: 8,
        _sockname: null,
        _pendingData: [
          {
            chunk: 'POST /Users/authenticatebyname HTTP/1.1\r\n' +
              'Accept: application/json, text/plain, */*\r\n' +
              'Content-Type: application/json\r\n' +
              'X-Emby-Authorization: MediaBrowser Client="Jellyfin Stremio Addon", Device="4f21e3b91aed", DeviceId="4f21e3b91aed", Version="1.0.0.0""\r\n' +
              'User-Agent: axios/1.6.6\r\n' +
              'Content-Length: 39\r\n' +
              'Accept-Encoding: gzip, compress, deflate, br\r\n' +
              'Host: localhost\r\n' +
              'Connection: keep-alive\r\n' +
              '\r\n',
            encoding: 'latin1',
            callback: [Function: nop]

          },
          {
            chunk: [Buffer [Uint8Array]],
            encoding: 'buffer',
            callback: [Function (anonymous)]

          },
          allBuffers: false
        ],
        _pendingEncoding: '',
        server: null,
        _server: null,
        timeout: 5000,
        parser: null,
        _httpMessage: [Circular *2],
        [Symbol(async_id_symbol)]: 275,
        [Symbol(kHandle)]: null,
        [Symbol(lastWriteQueueSize)]: 0,
        [Symbol(timeout)]: Timeout {
          _idleTimeout: -1,
          _idlePrev: null,
          _idleNext: null,
          _idleStart: 395,
          _onTimeout: null,
          _timerArgs: undefined,
          _repeat: null,
          _destroyed: true,
          [Symbol(refed)]: false,
          [Symbol(kHasPrimitive)]: false,
          [Symbol(asyncId)]: 274,
          [Symbol(triggerId)]: 0
        },
        [Symbol(kBuffer)]: null,
        [Symbol(kBufferCb)]: null,
        [Symbol(kBufferGen)]: null,
        [Symbol(shapeMode)]: true,
        [Symbol(kCapture)]: false,
        [Symbol(kSetNoDelay)]: true,
        [Symbol(kSetKeepAlive)]: true,
        [Symbol(kSetKeepAliveInitialDelay)]: 60,
        [Symbol(kBytesRead)]: 0,
        [Symbol(kBytesWritten)]: 0
      },
      [Symbol(kOutHeaders)]: [Object: null prototype] {
        accept: [ 'Accept', 'application/json, text/plain, */*' ],
        'content-type': [ 'Content-Type', 'application/json' ],
        'x-emby-authorization': [
          'X-Emby-Authorization',
          'MediaBrowser Client="Jellyfin Stremio Addon", Device="4f21e3b91aed", DeviceId="4f21e3b91aed", Version="1.0.0.0""'
        ],
        'user-agent': [ 'User-Agent', 'axios/1.6.6' ],
        'content-length': [ 'Content-Length', '39' ],
        'accept-encoding': [ 'Accept-Encoding', 'gzip, compress, deflate, br' ],
        host: [ 'Host', 'localhost' ]
      },
      [Symbol(errored)]: null,
      [Symbol(kHighWaterMark)]: 16384,
      [Symbol(kRejectNonStandardBodyWrites)]: false,
      [Symbol(kUniqueHeaders)]: null
    },
    _currentUrl: 'http://localhost/Users/authenticatebyname',
    [Symbol(shapeMode)]: true,
    [Symbol(kCapture)]: false
  },
  cause: Error: connect ECONNREFUSED 127.0.0.1:80
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1602:16) {
    errno: -111,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '127.0.0.1',

    port: 80
  }
}
Node.js v21.6.0

Thanks regard , if have docker-compose more easy.

**Comment by akarazniewicz** [comment]
Can You please reorder arguments?

`docker run -p 60421:60421 -e JELLYFIN_USER=plsharevme -e JELLYFIN_PASSWORD=1234567 -e JELLYFIN_SERVER=http://192.168.1.100:8096/ ghcr.io/akarazniewicz/stremio-jellyfin `

**Comment by ericvlog** [comment]
ohh....thanks i redo it came : 

Connecting to Jellyfin server: http://192.168.1.100:8096/ with username: plsharevme and password: 123456
(node:1) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `nodejs --trace-deprecation ...` to show where the warning was created)
Error caught while Jellyfin authentication, server response: '404' and data: '<empty>' (server: 'http://192.168.1.100:8096/' with username: 'plsharevme' and password: ' 123456')
Exiting. Please check your configuration and Jellyfin connection.

**Comment by akarazniewicz** [comment]
Yeah, now looks like You have wrong password (space at the beginning? ' 123456'). 404 means wrong username/password, but it's connectable.

**Comment by ericvlog** [comment]
it weird , i double checked password username is okay,......i even sign out and login back in jellyfin without problem.

**Comment by akarazniewicz** [comment]
No, thats jus warning from nodejs. 

Can You put Your password in braces when starting addon? JELLYFIN_PASSWORD="1234567"

**Comment by ericvlog** [comment]
ohhh......i know what cause the problem , is my  JELLYFIN_SERVER="http://192.168.1.100:8096/" ,.....cannot have "/" at end of url ...

but now new problem is 

Connecting to Jellyfin server: http://192.168.1.100:8096 with username: plsharevme and password: 123456
(node:1) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `nodejs --trace-deprecation ...` to show where the warning was created)
Successfully connected to Jellyfin server: http://192.168.1.100:8096. Happy streaming.
HTTP addon accessible at: http://127.0.0.1:60421/manifest.json

http://127.0.0.1:60421/manifest.json i copy and change to my host ip paste at stremio search addon, it say cannot find addon/collection from this URL , but i paste at browser url it does come out some language code .

some people say stremio addon need access from https if that true ?

**Comment by akarazniewicz** [comment]
This is internal docker address. You need to access IP of Your machine where You're running docker container. Port stays the same. So something like:

http://your_docker_host_ip:60421/manifest.json

**Comment by ericvlog** [comment]
yup ........my docker of ip is http://192.168.1.100:60421/manifest.json 

i open at browser :

{
    "id": "community.stremiojellyfin",
    "version": "1.0.0",
    "catalogs": [
        {
            "type": "movie",
            "id": "all",
            "extra": [
                {
                    "name": "skip",
                    "isRequired": true
                },
                {
                    "name": "search",
                    "isRequired": false
                }
            ]
        },
        {
            "type": "series",
            "id": "all",
            "extra": [
                {
                    "name": "skip",
                    "isRequired": true
                },
                {
                    "name": "search",
                    "isRequired": false
                }
            ]
        }
    ],
    "resources": [
        "catalog",
        "stream",
        "meta"
    ],
    "types": [
        "movie",
        "series"
    ],
    "name": "Jellyfin",
    "description": "Stremio Jellyfin integration"

but i paste it to stremio addon it say cannot find addon/collection from this URL

**Comment by CaptainMihawk** [comment]
I have the same error, the message i got is "failed to fetch" and the server url http://192.168.1.110:60421/manifest.json is accessible in the browser

**Comment by akarazniewicz** [comment]
Can You install it from browser app? web.stremio.com

**Comment by ericvlog** [comment]
![image](https://github.com/akarazniewicz/stremio-jellyfin/assets/64370408/8aeeeeb6-2072-4c0b-a117-f934dc76a1a5)


![image](https://github.com/akarazniewicz/stremio-jellyfin/assets/64370408/5e87d468-2308-4517-a975-5aaf1fee77a2)

![image](https://github.com/akarazniewicz/stremio-jellyfin/assets/64370408/5836bf17-46c0-41bc-b240-943e6329673f)

failed

**Comment by akarazniewicz** [comment]
Ah interesting. This may be somehow connected to cors. Will need to test it. In my case I have everything proxied with SSL - so this may be it. But unfortunately this is way, way more complicated setup.

**Comment by CaptainMihawk** [comment]
> Can You install it from browser app? web.stremio.com

it worked for me

**Comment by akarazniewicz** [comment]
You may have different network configuration (not sure if in @ericvlog case his browser and his addon IP are on the same subnet). Worked for me withouth SSL too, but again, I have everything on the same network.

Second thing is - I use Firefox, that may play some role here, too.

**Comment by ericvlog** [comment]
yup ......couse i hear before the addon(http) need access from local stremio "http" stremio , maybe i host own stremio server later for try.....anyway thanks for your help.

**Comment by ericvlog** [comment]
hello, i managed to installed by nginx proxy manager (https) and tested run well on **Android TV Stremio**.

but catalog no appear on android stremio ( only android tv stremio), and i tested on windows stremio also no catalog.

1 more suggestion, can we split the catalog by library jellyfin?

Anyway I do tested play on Android TV Stremio without any problem.! thanks

**Comment by alanmilinovic** [comment]
> Ah interesting. This may be somehow connected to cors. Will need to test it. In my case I have everything proxied with SSL - so this may be it. But unfortunately this is way, way more complicated setup.

I as well have everything proxied with SSL and am getting failed to fetch error 500.

Do I need to have jellyfin as well proxied and with SSL for this to work, or just stremio? So far I tried with proxied jellyfin with and without SSL.

This is how I proxied:
```
<VirtualHost *:20537>
	<Location "/jellyfin-stremio-addon/">
		ProxyPass http://localhost:60421/
		ProxyPassReverse http://localhost:60421/
	</Location>
</VirtualHost>
```

The URL looks like this:
`https://subdomain.domain.com:20537/jellyfin-stremio-addon/manifest.json`

The error message in docker:
``HTTP addon accessible at: http://127.0.0.1:60421/manifest.json
request for catalogs: movie all
TypeError: Cannot read properties of undefined (reading 'Id')
    at JellyfinApi.searchItems (file:///home/node/app/jellyfin.js:47:68)
    at file:///home/node/app/addon.js:32:49
    at AddonInterface.get (/home/node/app/node_modules/stremio-addon-sdk/src/builder.js:90:10)
    at /home/node/app/node_modules/stremio-addon-sdk/src/getRouter.js:23:3
    at Layer.handle [as handle_request] (/home/node/app/node_modules/router/lib/layer.js:93:5)
    at next (/home/node/app/node_modules/router/lib/route.js:161:13)
    at Route.dispatch (/home/node/app/node_modules/router/lib/route.js:121:3)
    at handle (/home/node/app/node_modules/router/index.js:532:11)
    at Layer.handle [as handle_request] (/home/node/app/node_modules/router/lib/layer.js:93:5)
    at /home/node/app/node_modules/router/index.js:300:15
request for catalogs: series all
TypeError: Cannot read properties of undefined (reading 'Id')
    at JellyfinApi.searchItems (file:///home/node/app/jellyfin.js:47:68)
    at file:///home/node/app/addon.js:32:49
    at AddonInterface.get (/home/node/app/node_modules/stremio-addon-sdk/src/builder.js:90:10)
    at /home/node/app/node_modules/stremio-addon-sdk/src/getRouter.js:23:3
    at Layer.handle [as handle_request] (/home/node/app/node_modules/router/lib/layer.js:93:5)
    at next (/home/node/app/node_modules/router/lib/route.js:161:13)
    at Route.dispatch (/home/node/app/node_modules/router/lib/route.js:121:3)
    at handle (/home/node/app/node_modules/router/index.js:532:11)
    at Layer.handle [as handle_request] (/home/node/app/node_modules/router/lib/layer.js:93:5)
    at /home/node/app/node_modules/router/index.js:300:15``

**Comment by alanmilinovic** [comment]
Just tested with proxing on top level, like this:
```
<VirtualHost *:20538>
    ServerName subdomain.domain.com

    SSLEngine on

    SSLCertificateFile      xxx
    SSLCertificateKeyFile   yyy

    ProxyPass / http://localhost:60421/
    ProxyPassReverse / http://localhost:60421/
</VirtualHost>
```

The URL in this case is:
`https://subdomain.domain.com:20538/manifest.json`

Same result unfortunately. Not sure why it is not working.

Maybe manifest URL must be without port? @akarazniewicz, can you please test it with port and SSL? I cannot test it as I have limitation in ports and 443 port is taken.

**Comment by ericvlog** [comment]
> Just tested with proxing on top level, like this:
> 
> ```
> <VirtualHost *:20538>
>     ServerName subdomain.domain.com
> 
>     SSLEngine on
> 
>     SSLCertificateFile      xxx
>     SSLCertificateKeyFile   yyy
> 
>     ProxyPass / http://localhost:60421/
>     ProxyPassReverse / http://localhost:60421/
> </VirtualHost>
> ```
> 
> The URL in this case is: `https://subdomain.domain.com:20538/manifest.json`
> 
> Same result unfortunately. Not sure why it is not working.
> 
> Maybe manifest URL must be without port? @akarazniewicz, can you please test it with port and SSL? I cannot test it as I have limitation in ports and 443 port is taken.

I use nginx proxy manager, It very easy to setup,.....my domain of addon https://stremio-jellyfin.mydomain.org .......it working quite well on android tv stremio.

**Comment by alanmilinovic** [comment]
> > Just tested with proxing on top level, like this:
> > ```
> > <VirtualHost *:20538>
> >     ServerName subdomain.domain.com
> > 
> >     SSLEngine on
> > 
> >     SSLCertificateFile      xxx
> >     SSLCertificateKeyFile   yyy
> > 
> >     ProxyPass / http://localhost:60421/
> >     ProxyPassReverse / http://localhost:60421/
> > </VirtualHost>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The URL in this case is: `https://subdomain.domain.com:20538/manifest.json`
> > Same result unfortunately. Not sure why it is not working.
> > Maybe manifest URL must be without port? @akarazniewicz, can you please test it with port and SSL? I cannot test it as I have limitation in ports and 443 port is taken.
> 
> I use nginx proxy manager, It very easy to setup,.....my domain of addon https://stremio-jellyfin.mydomain.org .......it working quite well on android tv stremio.

Yes, but I am limited with ports and my addon is https://subdomain.domain.com:20538/manifest.json, I think that stremio when accessing with SSL is not handling ports very good in general. I had some issues when streaming server URL is with port also, then video player is not working.

@ericvlog can you try to configure your addon with port and test please?

**Comment by ericvlog** [comment]
Why do u include port in your SSL ?.......

Local-ip:port forwarding to SSL 

Should be https://stremio-jellyfin.domain.org

Without port.

**Comment by ericvlog** [comment]
> > > Just tested with proxing on top level, like this:
> > > ```
> > > <VirtualHost *:20538>
> > >     ServerName subdomain.domain.com
> > > 
> > >     SSLEngine on
> > > 
> > >     SSLCertificateFile      xxx
> > >     SSLCertificateKeyFile   yyy
> > > 
> > >     ProxyPass / http://localhost:60421/
> > >     ProxyPassReverse / http://localhost:60421/
> > > </VirtualHost>
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > The URL in this case is: `https://subdomain.domain.com:20538/manifest.json`
> > > Same result unfortunately. Not sure why it is not working.
> > > Maybe manifest URL must be without port? @akarazniewicz, can you please test it with port and SSL? I cannot test it as I have limitation in ports and 443 port is taken.
> > 
> > I use nginx proxy manager, It very easy to setup,.....my domain of addon https://stremio-jellyfin.mydomain.org .......it working quite well on android tv stremio.
> 
> Yes, but I am limited with ports and my addon is https://subdomain.domain.com:20538/manifest.json, I think that stremio when accessing with SSL is not handling ports very good in general. I had some issues when streaming server URL is with port also, then video player is not working.
> 
> @ericvlog can you try to configure your addon with port and test please?

![IMG_20240313_165734.jpg](https://github.com/akarazniewicz/stremio-jellyfin/assets/64370408/e8ecd22e-1a43-4f07-8e09-c7b03b3a7e9e)

**Comment by alanmilinovic** [comment]
> Why do u include port in your SSL ?.......
> 
> Local-ip:port forwarding to SSL
> 
> Should be https://stremio-jellyfin.domain.org
> 
> Without port.

Because I have VPN on entire machine and use remote port forwarding on VPN backend where other subscribers are using ports as well and ports 80 or 443 are reserved. I didn't managed to forward port from tunnel to lan to be able to regularly forward in my router publicly.

**Comment by alanmilinovic** [comment]
I have an update, I managed to get rid of the port, but still same issue.

**Comment by alanmilinovic** [comment]
Last update from my side, I gave up on this plugin it is not working for me. So, I put Jellyfin on SSL public link without port and used it in docker compose file still not working.

**Comment by ericvlog** [comment]
> Last update from my side, I gave up on this plugin it is not working for me. So, I put Jellyfin on SSL public link without port and used it in docker compose file still not working.

U need SSL on add-ons configuration page no jellyfin.

**Comment by alanmilinovic** [comment]
Bro, tried all possible combinations, it is not working.

**Comment by alanmilinovic** [comment]
![Screenshot_20240316_123800_Chrome](https://github.com/akarazniewicz/stremio-jellyfin/assets/1752983/04248bc6-bda7-4f57-a5eb-5a105d7a20e0)

**Comment by ericvlog** [comment]
> ![Screenshot_20240316_123800_Chrome](https://github.com/akarazniewicz/stremio-jellyfin/assets/1752983/04248bc6-bda7-4f57-a5eb-5a105d7a20e0)
> 

U have try on android TV(stremio) ?

No phone & windows stremio

**Comment by alanmilinovic** [comment]
> > ![Screenshot_20240316_123800_Chrome](https://private-user-images.githubusercontent.com/1752983/313401614-04248bc6-bda7-4f57-a5eb-5a105d7a20e0.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA1OTE3NjMsIm5iZiI6MTcxMDU5MTQ2MywicGF0aCI6Ii8xNzUyOTgzLzMxMzQwMTYxNC0wNDI0OGJjNi1iZGE3LTRmNTctYTVlYi01YTEwNWQ3YTIwZTAuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTZUMTIxNzQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjA1NTVkNjE3NzQ1ZjZkYzE4YzFlNTAwYjIzMGY1MTUzNmYwYWJlNDNlYTFmNzVkODA2YjNlN2UxYTE5MGUxOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Fi8j3nBKTYvzBsMsIrw4w3dZaA2wkLcevpfjyGesyFA)
> 
> U have try on android TV(stremio) ?
> 
> No phone & windows stremio

Yes, I did.

**Comment by drizzt09** [comment]
any update on this?

In webversion i get error menioned above:
![image](https://github.com/akarazniewicz/stremio-jellyfin/assets/6366965/c994f823-7685-4c86-998c-86bbbc15989c)

yes it is http but I do not want to start going down the https route if it isnt going to solve the issue.
looks like its tried above but never solved.

i did install Nginx proxy manager but i do not know how to set it up and as mentioned, above it didnt really help.

**Comment by alanmilinovic** [comment]
It is not working with https or http.

**Comment by ericvlog** [comment]
> It is not working with https or http.

Now it is working well, sorry no feedback here, but if I can manually add library as catalog it more better...now it mixes all library into movies catalog

**Comment by drizzt09** [comment]
> > It is not working with https or http.
> 
> Now it is working well, sorry no feedback here, but if I can manually add library as catalog it more better...now it mixes all library into movies catalog

So what steps you performed to get it working? Beyond where I'm stuck?

And the images show it split by library. TV and movie.

**Comment by drizzt09** [comment]
> > It is not working with https or http.
> 
> Now it is working well, sorry no feedback here, but if I can manually add library as catalog it more better...now it mixes all library into movies catalog

@ericvlog 
what did you do to get it working? I am stuck on no manifest

**Comment by ericvlog** [comment]
> > > It is not working with https or http.
> > 
> > Now it is working well, sorry no feedback here, but if I can manually add library as catalog it more better...now it mixes all library into movies catalog
> 
> @ericvlog 
> what did you do to get it working? I am stuck on no manifest

The most important step is "Https" ssl.....

**Comment by alanmilinovic** [comment]
People it doesn't work. I tried everything.

**Comment by drizzt09** [comment]
> > > > It is not working with https or http.
> > > 
> > > Now it is working well, sorry no feedback here, but if I can manually add library as catalog it more better...now it mixes all library into movies catalog
> > 
> > @ericvlog 
> > what did you do to get it working? I am stuck on no manifest
> 
> The most important step is "Https" ssl.....

I have https ssl.
The problem is no manifest.json like gets created. When I manually crested one it still doesn't work as I don't know if it is made correctly. The dev has just disappeared.

**Comment by alanmilinovic** [comment]
Yes, just forget it and move on.

**Comment by Logikoz** [comment]
> ![image](https://github.com/akarazniewicz/stremio-jellyfin/assets/64370408/8aeeeeb6-2072-4c0b-a117-f934dc76a1a5)
> 
> ![image](https://github.com/akarazniewicz/stremio-jellyfin/assets/64370408/5e87d468-2308-4517-a975-5aaf1fee77a2)
> 
> ![image](https://github.com/akarazniewicz/stremio-jellyfin/assets/64370408/5836bf17-46c0-41bc-b240-943e6329673f)
> 
> failed

same problem here...

solution? @akarazniewicz

**Comment by TsvetanTsvetanov97** [comment]
Hi guys, in my case this issue

<img width="2506" height="1322" alt="Image" src="https://github.com/user-attachments/assets/8c9dda74-ab2d-43e9-8a7d-d4c0c9278b68" />

 was caused by chrome blocking the script

<img width="539" height="421" alt="Image" src="https://github.com/user-attachments/assets/ce48632a-4776-4784-8551-4efba81daf69" />

, after clicking "Load usafe scripts", it worked!

**Comment by alanmilinovic** [comment]
> Hi guys, in my case this issue
> 
> <img alt="Image" width="2000" height="1322" src="https://private-user-images.githubusercontent.com/53956542/488034161-8c9dda74-ab2d-43e9-8a7d-d4c0c9278b68.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc1NzA2ODMsIm5iZiI6MTc1NzU3MDM4MywicGF0aCI6Ii81Mzk1NjU0Mi80ODgwMzQxNjEtOGM5ZGRhNzQtYWIyZC00M2U5LThhN2QtZDRjMGM5Mjc4YjY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTExVDA1NTk0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFhY2U1N2QwYjY2MjA0MDQyMjc2MDhhZjZiMGZkYWRlZDUwN2Y5YjIwYzRjM2E3Y2E4MzQ4ZmFlMGU4YjUxOWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8RwGteDI9_6kQKDucha6zr3-VQzcVqzpGF9c-ToTEmo">
> was caused by chrome blocking the script
> 
> <img alt="Image" width="539" height="421" src="https://private-user-images.githubusercontent.com/53956542/488034788-ce48632a-4776-4784-8551-4efba81daf69.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc1NzA2ODMsIm5iZiI6MTc1NzU3MDM4MywicGF0aCI6Ii81Mzk1NjU0Mi80ODgwMzQ3ODgtY2U0ODYzMmEtNDc3Ni00Nzg0LTg1NTEtNGVmYmE4MWRhZjY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTExVDA1NTk0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkOGIxMWIwZmI4ODE4MzQ2YjY2ODYyNmNlYzdmZGQ4ZGFhYTVmZDZiNjE2NTg2ZTE3Yzk1NWEwYTY2MTliYjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3WrvWthhZ16NVpiohE89jmxpT3WygN9NFB2ydyKkV_M">
> , after clicking "Load usafe scripts", it worked!

I am not getting this popup and still have the same error 500.

**Comment by TsvetanTsvetanov97** [comment]
Hi, here is a quick tutorial how to make the plugin connection secure, and also make it publicly available. Which should solve the issue, with loading the plugin in Stremio. WARNING, I'm no networking specialist by any means, opening ports can be security issue, do this steps on your own risk.


**1.** Create your two domains(one for Jellyfin, one for the plugin) i used FreeDNS, because it's free and it's pretty simple, if your router has dynamic IP, you should use their DDNS application, i have static one, so i had no need to use the DDNS, i just created a domaint and assigned it to the ip of my router.

**NOTE:** If you want you can try to setup you local DNS, this will limit the access to the plugin and the server to the local network, but it will be more secure, as there will be no need to ports on the router. If you decide to proceed this way, you can skip steps **2 and 3**,

**2** Forward the ports 443 and 80 to the machine your docker is running on. This is done in the settings on your router.

**3** Open ports 443 and 80 in Inbound Rules in Windows Defender Firewall

**4** Setup Nginx Proxi Manager, We will use it as reverse proxi to catch the requests comming to the machine and forward them to Jellyfin and the Stremio plugin, and also encrypt them.

        You can use this file to compose the nginx container.                            
[docker-compose.yml](https://github.com/user-attachments/files/22314989/docker-compose.yml)
          You just need to save the file somewhere, and run "docker-compose up -d" command.

*5* Access Nginx Proxi Manager from http://localhost:81/login, default username: admin@example.com, pass: changeme

**6** Go to Hosts -> Proxy Hosts -> Add Proxy Host -> Fill the fields:
       -- Domain Names: The domain for the plugin you created in the first step
       -- Scheme : HTTP
       --IP: The IP of the machine hasting the conatianer
       -- Port: the port of the plugin, the default one is 60421
       --Click block common exploits
   -> Go to SSL tab
       -- Select "Request New SSL Certificate"
       -- Select "Force SSL"
      -- Click Save
 -> Do these steps again, but for the Jellyfin domain with the Jellyfin port.

**NOTE:** There are some options to limit the access by IP/MAC, but I haven't figured them out yet.


**7** Now you must rebuild the plugin container, but use the domain of Jellyfin instead of the IP of the machine.

@alanmilinovic Hi, check this please.

**Comment by alanmilinovic** [comment]
I will, thanks for the detailed instructions. So far I was using tunnel from cloudflare with https to expose Jellyfin and Stremio as well as the plugin itself, and it was all publicly reachable. But what I can try is to use ddns instead and play a bit more.
